src/isolate.rs

```diff
@@ -289,6 +295,25 @@ impl Isolate {
     }
   }
 
+  /// Return data that was previously attached to the isolate snapshot via
+  /// SnapshotCreator, and removes the reference to it.
+  /// Repeated call with the same index return None.
+  pub unsafe fn get_data_from_snapshot_once<'s, T>(
+    &self,
```

Comment:
Yeah so this function can't work because the caller would need to give you two (mutable) references to the same Isolate. One in the self param (which should be `&mut self`) and the other one emedded in the HandleScope.

So I would implement this method on the scope instead.

```rust
  pub unsafe fn get_data_from_snapshot_once<'s, T>(
    self: &mut HandleScope<'s, ()>,
    index: usize,
  ) -> Option<Local<'s, T>>
  where
    for<'l> Local<'l, Data>: From<Local<'l, T>>,
  {
    self
      .cast_local(|sd| v8__Isolate__GetDataFromSnapshotOnce(sd.get_isolate_ptr(), index))
      .map(Local::cast)
  }
```